pkg/util/tracing/collector/collector.go

```diff
@@ -149,9 +155,19 @@ func (i *Iterator) Next() {
 	}
 }
 
+// NodeRecording contains the node and a Recording i.e. span recordings for all
+// spans rooted at a fixed root span.
+type NodeRecording struct {
+	NodeID    roachpb.NodeID
+	Recording tracing.Recording
+}
+
 // Value returns the current value pointed to by the Iterator.
-func (i *Iterator) Value() tracingpb.RecordedSpan {
-	return i.recordedSpans[i.recordedSpanIndex]
+func (i *Iterator) Value() NodeRecording {
+	return NodeRecording{
+		NodeID:    i.curNode,
```

Comment:
Why return the NodeID at all? Do we expect to use it? I imagined the value type would've simply been a trace.Recording instead.

Reply:
The way I envision it is that in  https://github.com/cockroachdb/cockroach/pull/66679, the vtable is going to store rows with the node_id. This will allow the CLI wrapper built on top of to dump all the inflight trace spans of a particular node in a single file.
In the case of jobs, you then get a per node view of job execution.

Reply:
Could return two values instead of introducing a wrapper type: (roachpb.NodeID, tracing.Recording)